Add option for increased output verbosity

bandre-ucar opened the pull request:
Implement the intermediate verbosity level for status and checkout.

User interface changes?: Yes
The command line arg '--verbose' can now be used 0, 1, or 2 times. Each time it is used increases the verbosity level of the output. The previous --verbose output is now obtained by using '--verbose --verbose'.

Fixes:
* GH-48 : provides additional information about what is checked out when running 'checkout_externals --status --verbose'
* GH-46, GH-49 : provides the git and svn commands being run via 'checkout_externals --verbose'

Testing:
  unit tests: python2 pass
  system tests: python2 pass, one skip
  manual testing: manually verified new output for cesm and standalone checkouts, empty, modified and out of sync output.

bandre-ucar committed:
Rename verbose status functions.

Rename the verbose_status functions to verbose_status_dump in preparation for
adding a new verbose status output level.

Testing:

  python2 make test - all tests pass, on test skipped
  manual testing - status and status verbose work as expected.

bandre-ucar committed:
Implement intermediate verbosity level

Implement the intermediate verbosity level for status and checkout.

Testing:

  make test - python2 - all tests pass, one skip
  manual testing of --verbose checkout and status.

billsacks commented on manic/checkout.py:
```diff
@@ -216,9 +217,10 @@ def commandline_arguments(args=None):
                         help='Output additional information to '
-                        'the screen and log file.')
+                        'the screen and log file. This flag can be used '
+                        'multiple times.')
```
Maybe be more explicit here, like, "This flag can be used up to 2 times, increasing the verbosity level each time."

billsacks replied:
Actually, maybe it only is effective the second time for '--status', in which case this could say, "For --status, this flag can be used a second time, increasing the verbosity level each time."

bandre-ucar committed:
Refine verbose output

Refine the verbose output based on PR feedback from Bill Sacks.

* Verbose output is cumulative as the verbose count increases.

* checkout and status have similar behavior

* dump level status output is indented consistently.

Testing:
  make test - all tests pass
  manual testing of cesm and standalone repos with new verbose checkout and status output.

billsacks commented on manic/checkout.py:
```diff
@@ -216,9 +216,11 @@ def commandline_arguments(args=None):
                         help='Output additional information to '
-                        'the screen and log file.')
+                        'the screen and log file. For --status, this flag '
+                        'can be used a second time, increasing the verbosity'
```
need a space after verbosity

billsacks replied:
This is addressed in https://github.com/bandre-ucar/manage_externals/pull/2 if it is accepted.

billsacks committed:
If we're already in sync, do not show output from the checkout command

However, specifying '--verbose --verbose' will still show output from
the checkout command even if we're already in sync.

This implementation feels like a bit of a hack. On the other hand, I can
convince myself that it makes sense as a general rule to decrease the
verbosity level of the checkout if we're in sync.
